### PR TITLE
Jtwig Extension for Converting bytes to human readable format

### LINKED ISSUE #1529: Create a jtwig extension which converts byte size into human readable size
# Task Description

We need to create a jTwig extension which is capable for converting the byte size into a human readable string. Although the `byte size` should in most cases be a `Long` this is going to be used in the templates and the value might sometimes also be a `String`. If the value cannot be properly converted - it should return the same value.

The extension should be placed a new package called `org.carlspring.strongbox.jtwig.extensions`.

Example usage in the template:

```
# find directoryListing.twig.html in `strongbox-web-core

<td>{{ file.size | default("-") | humanReadableSize  }}</td>
```

To check your code:
1. [Start a Strongbox instance](https://strongbox.github.io/developer-guide/building-strongbox-using-strongbox-instance.html#starting-a-strongbox-instance) 
2. Download an artifact by going to `http://localhost:48080/storages/storage-common-proxies/carlspring/org/carlspring/maven/derby-maven-plugin/1.10/derby-maven-plugin-1.10.jar`
3. Browse the artifact by going to `http://localhost:48080/api/browse/storage-common-proxies/carlspring/org/carlspring/maven/derby-maven-plugin/1.10/`
4. The `Size` column should show the size in human readable format like `1.0 MB` instead of the `long` it usually shows.

# Useful Links

* [jTwig docs](http://jtwig.org/documentation/reference/web/spring-boot)
* [Byte size into human readable](https://stackoverflow.com/a/3758880/393805)

# Help
* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @fuss86
  * @sbespalov
  * @steve-todorov

### PR DESCRIPTION
**Pull request by anki2189**
# Pull Request Description

This pull request closes #1529 

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [ ] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

### CODE REVIEW

**Comment by anki2189** [comment]
image

**Comment by anki2189** [comment]
@carlspring Pls review the change.

**Comment by carlspring** [comment]
Thanks for the pull request, @anki2189 !

@steve-todorov : Please, review!

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/jtwig/extensions/HumanReadableSizeConversionFunction.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            // Need to return same value of bytes
+        }
+        return byteSize;
+    }
+
+    private String getHumanReadableSize(long bytes)
+    {
+        int unit = 1000;
+        if (bytes < unit)
+            return bytes + " B";
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/jtwig/extensions/HumanReadableSizeConversionFunction.java]: Cosmetic: Please, always use braces around conditions and loops. Thanks! :)

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/jtwig/extensions/HumanReadableSizeConversionFunction.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    @Override
+    public String name()
+    {
+        return "humanReadableSize";
+    }
+
+    @Override
+    public Collection<String> aliases()
+    {
+        return Collections.unmodifiableList(Lists.newArrayList("readableSize"));
```
Comment (steve-todorov) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/jtwig/extensions/HumanReadableSizeConversionFunction.java]: No need to use guava here (and the import) :)
```suggestion
        return Collections.unmodifiableList(Arrays.asList("readableSize"));
```

Comment (anki2189) [reply strongbox-web-core/src/main/java/org/carlspring/strongbox/jtwig/extensions/HumanReadableSizeConversionFunction.java]: This is what I was thinking to initially. :)

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/jtwig/extensions/HumanReadableSizeConversionFunction.java (hunk 1 above)
Comment (anki2189) [reply strongbox-web-core/src/main/java/org/carlspring/strongbox/jtwig/extensions/HumanReadableSizeConversionFunction.java]: Noted

**Comment by carlspring** [comment]
@anki2189 ,

Would you mind using `git rebase`, instead of `git merge` when synchronizing your fork? This helps keep the history cleaner to read. For more details, you can check [here](https://strongbox.github.io/developer-guide/git/rebase-vs-merge.html).

**Comment by anki2189** [comment]
Noted @carlspring Will do rebase from next PR.

**Comment by carlspring** [comment]
Great! Thanks! :)

**Comment by carlspring** [comment]
Thanks for these fixes! :)

**Pull request merged**